cli/core/docs/cli/cli-architecture.md

```diff
@@ -159,9 +159,9 @@ tanzu mission-control cluster list
 
 ## Plugin Discovery Sources
 
-NOTE: This is applicable only if [context-aware plugin discovery](/docs/design/context-aware-plugin-discovery-design.md) is enabled within Tanzu CLI.
+NOTE: This is applicable only if [context-aware plugin discovery](../design/context-aware-plugin-discovery-design.md) is enabled within Tanzu CLI.
 
-Discovery is the interface to fetch the list of available plugins, their supported versions and how to download them either standalone or scoped to a context(server). E.g., the CLIPlugin API in a management cluster, OCI based plugin discovery for standalone plugins, a similar REST API and a manifest file in GCP based discovery, etc. (API is defined [here](apis/config/v1alpha1/clientconfig_types.go#L111-L187)) Unsupported plugins and plugin versions are not returned by the interface. Having a separate interface for discovery helps to decouple discovery (which is usually tied to a server or user identity) from distribution (which can be shared).
+Discovery is the interface to fetch the list of available plugins, their supported versions and how to download them either standalone or scoped to a context(server). E.g., the CLIPlugin API in a management cluster, OCI based plugin discovery for standalone plugins, a similar REST API and a manifest file in GCP based discovery, etc. (API is defined [here](../../../../apis/config/v1alpha1/clientconfig_types.go)) Unsupported plugins and plugin versions are not returned by the interface. Having a separate interface for discovery helps to decouple discovery (which is usually tied to a server or user identity) from distribution (which can be shared).
```

Comment:
nit: I might need to update `../../../../apis/config/v1alpha1/clientconfig_types.go` path as part of PR #3375 👍